packages/did/src/Did.utils.ts

```diff
@@ -161,31 +160,38 @@ const multicodecReversePrefixes: Record<DidVerificationMethodType, number> = {
 export function multibaseKeyToDidKey(
   publicKeyMultibase: VerificationMethod['publicKeyMultibase']
 ): DecodedVerificationMethod {
-  if (!publicKeyMultibase.startsWith('z')) {
-    throw new SDKErrors.DidError(`invalid format for '${publicKeyMultibase}'`)
-  }
-  const decodedMulticodecPublicKey = base58Decode(publicKeyMultibase.slice(1))
-  const [keyTypeFlag, publicKey] = [
-    decodedMulticodecPublicKey.subarray(0, 1)[0],
-    decodedMulticodecPublicKey.subarray(1),
-  ]
-  const [keyType, expectedPublicKeyLength] = multicodecPrefixes[keyTypeFlag]
+  const { keyBytes, prefix } = decodeBase58BtcMultikey(publicKeyMultibase)
+
+  const [keyType, expectedPublicKeyLength] = multicodecPrefixes[prefix]
   if (keyType === undefined) {
     throw new SDKErrors.DidError(
       `Cannot decode key type for multibase key "${publicKeyMultibase}".`
     )
   }
-  if (publicKey.length !== expectedPublicKeyLength) {
+  if (keyBytes.length !== expectedPublicKeyLength) {
     throw new SDKErrors.DidError(
-      `Key of type "${keyType}" is expected to be ${expectedPublicKeyLength} bytes long. Provided key is ${publicKey.length} bytes long instead.`
+      `Key of type "${keyType}" is expected to be ${expectedPublicKeyLength} bytes long. Provided key is ${keyBytes.length} bytes long instead.`
     )
   }
   return {
     keyType,
-    publicKey,
+    publicKey: keyBytes,
   }
 }
 
+// TODO: This could also be exposed in a new release candidate of the `@kiltprotocol/jcs-data-integrity-proofs-common` package.
+function multibase58BtcKeyBytesEncoding(
+  key: Uint8Array,
+  keyPrefix: number
+): Base58BtcMultibaseString {
+  const encodeVarint =
+    typeof varint.encode === 'function' ? varint.encode : varint.default.encode
+  const varintEncodedPrefix = encodeVarint(keyPrefix)
+  const prefixedKey = Uint8Array.from([...varintEncodedPrefix, ...key])
```

Comment:
polkadot/utils has the u8aConcat function for that, I think

Reply:
Yup! https://github.com/KILTprotocol/sdk-js/pull/872/commits/bd5df5fec8b12da2f99bd204dee3f23f15954a27